lib/pacemaker/pcmk_sched_location.c

```diff
@@ -327,13 +327,27 @@ unpack_rsc_location(xmlNode *xml_obj, pcmk_resource_t *rsc,
         /* This loop is logically parallel to pe_evaluate_rules(), except
          * instead of checking whether any rule is active, we set up location
          * constraints for each active rule.
+         *
+         * @COMPAT When we can break backward compatibility, limit location
+         * constraints to a single rule, for consistency with other contexts.
+         * Since a rule may contain other rules, this does not prohibit any
+         * existing use cases.
```

Comment:
The existing documentation has numerous errors and inconsistencies with the code and schema, it will lead you astray :)

We only allow (via the schema) a single toplevel rule everywhere except location constraints, where they do not behave as documented. A single toplevel rule may include any number of subrules, so there's no point in allowing multiple at the top, and dropping support will let us simplify the code.